src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php

```diff
@@ -214,4 +228,20 @@ private function generateAlpha2ToAlpha3Mapping(ArrayAccessibleResourceBundle $me
 
         return $alpha2ToAlpha3;
     }
+
+    private function generateAlpha3ToAlpha2Mapping(ArrayAccessibleResourceBundle $metadataBundle): array
+    {
+        $alpha3ToAlpha2 = [];
+
+        foreach ($metadataBundle['alias']['language'] as $alias => $data) {
+            $language = $data['replacement'];
+            if (2 === \strlen($language) && 3 === \strlen($alias) && 'overlong' === $data['reason']) {
+                $alpha3ToAlpha2[$alias] = $language;
```

Comment:
This is duplication of the code in `generateAlpha2ToAlpha3Mapping`. If that function could return 2 arrays instead of one, we would avoid this code duplication, and also the inefficency of iterating all the alises twice.

But I guess the efficency of the code is not an issue since this is in the generator, but I still do not like the code duplication.

Reply:
`generateAlpha2ToAlpha3Mapping` already exists in 3.4, i prefer to not cause any conflicts

agree the "less efficient" code isnt really an issue for generating, which is still fast IMHO.

Given the Independence nature of both mappings, i actually prefer the separation. 

```php
if (2 === \strlen($language) && 3 === \strlen($alias) && 'overlong' === $data['reason']) {
```

this line could be moved to a static helper, sure. But im not sure it's worth it, only to block ourselves with in the future. And again we'd be causing conflicts with 3.4